pkg/documents/storage/bucket.go

```diff
@@ -24,40 +26,51 @@ import (
 
 type BucketFunc func() (*blob.Bucket, error)
 
-func GetBucket(bucketName, cloudProvider string, FileblobSigner *fileblob.URLSignerHMAC) BucketFunc {
+func GetBucket(logger log.Logger, bucketName, cloudProvider string, FileblobSigner *fileblob.URLSignerHMAC) BucketFunc {
 	return func() (*blob.Bucket, error) {
 		ctx, cancelFn := context.WithTimeout(context.TODO(), 10*time.Second)
 		defer cancelFn()
 
-		if bucketName == "" || cloudProvider == "" {
-			return nil, fmt.Errorf("storage: missing BUCKET_NAME=%s and/or CLOUD_PROVIDER=%s", bucketName, cloudProvider)
+		if bucketName == "" {
+			return nil, errors.New("missing document bucket")
 		}
-		return openBucket(ctx, bucketName, cloudProvider, FileblobSigner)
+		if cloudProvider == "" {
+			return nil, errors.New("missing documents cloud provider")
+		}
+
+		return openBucket(ctx, logger, bucketName, cloudProvider, FileblobSigner)
 	}
 }
 
 // openBucket returns a Go Cloud Development Kit (Go CDK) Bucket object which can be used to read and write arbitrary
 // blobs from a cloud provider blob store. Checkout https://gocloud.dev/ref/blob/ for more details
-func openBucket(ctx context.Context, bucketName, cloudProvider string, FileblobSigner *fileblob.URLSignerHMAC) (*blob.Bucket, error) {
+func openBucket(ctx context.Context, logger log.Logger, bucketName, cloudProvider string, FileblobSigner *fileblob.URLSignerHMAC) (*blob.Bucket, error) {
 	switch strings.ToLower(cloudProvider) {
 	case "aws":
-		return awsBucket(ctx, bucketName)
+		return awsBucket(ctx, logger, bucketName)
 	case "file":
-		return fileBucket(ctx, bucketName, FileblobSigner)
+		return fileBucket(ctx, logger, bucketName, FileblobSigner)
 	case "gcp":
-		return gcpBucket(ctx, bucketName)
+		return gcpBucket(ctx, logger, bucketName)
 	default:
 		return nil, fmt.Errorf("invalid cloud provider: %s", cloudProvider)
 	}
 }
 
-func awsBucket(ctx context.Context, bucketName string) (*blob.Bucket, error) {
+func awsBucket(ctx context.Context, logger log.Logger, bucketName string) (*blob.Bucket, error) {
 	c := &aws.Config{
 		Region:      aws.String(os.Getenv("AWS_REGION")),
 		Credentials: credentials.NewEnvCredentials(), // reads AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY
 	}
 	s := session.Must(session.NewSession(c))
-	return s3blob.OpenBucket(ctx, s, bucketName, nil)
+
+	bucket, err := s3blob.OpenBucket(ctx, s, bucketName, nil)
+	if err != nil {
+		logger.Log("storage", fmt.Sprintf("ERROR creating %s gcp bucket: %v", bucketName, err))
```

Comment:
I think `gcp` should be `aws` in the error message.

And I recommend returning early here:

```go
if err != nil {
  // log err
  return nil, err
}
// log success msg
return bucket, nil

Reply:
Yea. I'll clean it up.